lib/main/transcriptionService.ts

```diff
@@ -1,380 +1,213 @@
 import { grpcClient } from '../clients/grpcClient'
-import mainStore from './store'
-import { STORE_KEYS } from '../constants/store-keys'
 import log from 'electron-log'
-import { AudioChunkSchema, ItoMode } from '@/app/generated/ito_pb'
-import { create } from '@bufbuild/protobuf'
-import { InteractionsTable } from './sqlite/repo'
-import { v4 as uuidv4 } from 'uuid'
+import { ItoMode } from '@/app/generated/ito_pb'
 import { BrowserWindow } from 'electron'
 import { traceLogger } from './traceLogger'
+import { AudioStreamManager } from './audio/AudioStreamManager'
+import { InteractionManager } from './interactions/InteractionManager'
+import { WindowMessenger } from './messaging/WindowMessenger'
+import { TextInserter } from './text/TextInserter'
+import { getCursorContext } from '../media/selected-text-reader'
+import { grammarRulesService } from './grammar/GrammarRulesService'
 
 export class TranscriptionService {
-  private isStreaming = false
```

Comment:
As much as possible, I tried turning the transcriptionService into an orchestration class that handles audio, interactions, the grpc client, inserting text, etc. while trying to have the detailed logic in each specific class.

I don't think this class or the more focused classes are perfect, but I think this is generally an improvement. I'm hoping we can isolate changes more and have less merge conflicts.